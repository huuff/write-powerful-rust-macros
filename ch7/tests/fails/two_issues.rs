struct Person {
    age: u32,
    name: String,
}

#[panic_to_result_macro::panic_to_result]
fn create_person_with_two_issues(name: String, age: u32) -> Result<Person, String> {
    if age > 30 {
        panic!();
    }
    Ok(Person { name, age })
}
