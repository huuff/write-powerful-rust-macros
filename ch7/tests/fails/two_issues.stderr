error: this macro can only be applied to a function that does not yet return a Result. Signature: Result < Person, String >
 --> tests/fails/two_issues.rs:7:61
  |
7 | fn create_person_with_two_issues(name: String, age: u32) -> Result<Person, String> {
  |                                                             ^^^^^^^^^^^^^^^^^^^^^^

error: panic needs a message!

         = help: try to add a message
         = note: we will add the message to Result's Err

 --> tests/fails/two_issues.rs:9:9
  |
9 |         panic!();
  |         ^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
  --> tests/fails/two_issues.rs:12:2
   |
12 | }
   |  ^ consider adding a `main` function to `$DIR/tests/fails/two_issues.rs`
